internal/elasticsearch/common_test.go

```diff
@@ -564,6 +565,28 @@ func TestPodSpecHasTaintTolerations(t *testing.T) {
 	}
 }
 
+func TestElasticSearchSecurityContext(t *testing.T) {
+	podTemplate := newPodTemplateSpec(context.Background(), log.NewLogger("common-testing"), "test-node-name", "test-cluster-name", "test-namespace-name", api.ElasticsearchNode{}, api.ElasticsearchNodeSpec{}, map[string]string{}, map[api.ElasticsearchNodeRole]bool{}, nil, LogConfig{})
+
+	if !*podTemplate.Spec.SecurityContext.RunAsNonRoot {
+		t.Errorf("SecurityContext.RunAsNonRoot: expected true, actual %v", *podTemplate.Spec.SecurityContext.RunAsNonRoot)
+	}
```

Comment:
nit. It is possible to use `pointer.BoolDeref()` or `pointer.BoolEqual()` here instead of plain dereferencing.